src/qrb.c

```diff
@@ -53,3 +56,122 @@ bool get_qrb_for_locator(const char *locator, double *range, double *bearing) {
     return RIG_OK == qrb(my.Long, my.Lat, -dest_long, dest_lat,
 			 range, bearing);
 }
+
+
+/* Hamlib rotator control */
+
+int init_tlf_rot(void) {
+    int retcode;		/* generic return code from functions */
+
+    /*
+     * allocate memory, setup & open port
+     */
+    my_rot = rot_init((rot_model_t) myrot_model);
+
+    if (!my_rot) {
+	shownr("Unknown rotator model", myrot_model);
+	return -1;
+    }
+
+    if (rotportname == NULL || strlen(rotportname) == 0) {
+	showmsg("Missing rot port name!");
+	return -1;
+    }
+
+    rotportname[strlen(rotportname) - 1] = '\0';	// remove '\n'
+    strncpy(my_rot->state.rotport.pathname, rotportname,
+	    TLFFILPATHLEN - 1);
+
+    my_rot->state.rotport.parm.serial.rate = serial_rate;
+
+    // parse ROTCONF parameters
+    if (parse_rotconf() < 0) {
+	return -1;
+    }
+
+    retcode = rot_open(my_rot);
+
+    if (retcode != RIG_OK) {
+	TLF_LOG_WARN("rot_open: %s", rigerror(retcode));
+	return -1;
+    }
+
+    return 0;
+}
+
+void rotate_to_qrb() {
+    double bearing;
+    double range;
+    int retcode;		/* generic return code from functions */
+    int pfx_index = getctydata_pfx(current_qso.call);
+
+    prefix_data *pfx = prefix_by_index(pfx_index);
+
+    if (pfx->dxcc_ctynr > 0) {
+
+	if (pfx_index != my.countrynr && 0 == get_qrb(&range, &bearing)) {
+
+	    pthread_mutex_lock(&tlf_rot_mutex);
```

Comment:
do we need a mutex for rot?
rig controls are called from both threads (main + background), but rot looks to be called from main only.

another question: does `rot_set_position()` just send a message to the rotator and doesn't wait for it to actually reach to requested position?

Reply:
We probably don't need the mutex now, but I would think it's safer to include it. New code might violate the assumption.

set_position just starts the rotation. You wouldn't want the rest of the program to halt for a minute while the antenna is turning...

Thanks for the review, will fix the variable init later.

Reply:
rig control and rotator control are talking to different devices, so calling functions at the same time is not a problem.

We could use the same mutex, but... this feels wrong.

Reply:
Fwiw the next step would be to read out the bearing and show it in the UI (I get reminded about this every time I'm using the station from remote), there the mutex might already be required.